include/constants/battle_config.h

```diff
@@ -252,6 +252,7 @@
 // Other settings
 #define B_DOUBLE_WILD_CHANCE        0     // % chance of encountering two Pokémon in a Wild Encounter.
 #define B_MULTI_BATTLE_WHITEOUT     GEN_8 // In Gen4+, multi battles end when the Player and also their Partner don't have any more Pokémon to fight.
+#define B_EVOLUTION_AFTER_WHITEOUT  TRUE  // If set to TRUE, Pokemon that qualify for evolution after battle will evolve even if the player loses.
```

Comment:
Seeing as the evolution after whiteout was introduced in Gen 6, I'm thinking this should be changed to using Generation constants.